backend/src/announcement/announcement.service.ts

```diff
@@ -36,42 +41,77 @@ export class AnnouncementServices {
     }
   }
   public async createAnnouncement(
+    user: User,
     data: AnnouncementDTO
   ): Promise<AnnouncementDTO> {
     try {
       const announcement = await this.prisma.announcement.create({
         data,
       });
+      this.eventEmitter.emit('create.logs', {
```

Comment:
Not sure if using `EventEmitter` is the most efficient way in implementing this feature(activity logs). >:3

Reply:
this will trigger the `create.logs` event listener on the `activity-log-service.ts` file to create the activity log for the CRUD operation that is being executed. >:3